packages/renderer/src/Route.spec.ts

```diff
@@ -0,0 +1,93 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render, screen } from '@testing-library/svelte';
+import { createRouteObject } from 'tinro/dist/tinro_lib';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import RouteSpec from '/@/RouteSpec.svelte';
+import RouteWithRequestSpec from '/@/RouteWithRequestSpec.svelte';
+
+vi.mock(import('tinro/dist/tinro_lib'));
+
+beforeEach(() => {
+  vi.resetAllMocks();
+});
+
+test('renders the route with request parser', async () => {
+  vi.mocked(createRouteObject).mockImplementation(() => {
+    return {
+      update: vi.fn(),
+    };
+  });
+  const myParser = (request: {
+    query: Record<string, string>;
+    params: Record<string, string>;
+  }): { id: number; label: string } => {
+    return {
+      id: +request?.params?.id,
```

Comment:
looks like you read some TS features, I've not seen much usage of this in the code yet :-)